pkg/kapp/cmd/app/delete.go

```diff
@@ -149,6 +151,13 @@ func (o *DeleteOptions) Run() error {
 		return err
 	}
 
+	defer func() {
+		_, numDeleted, _ := app.GCChanges(o.DeleteFlags.AppChangesMaxToKeep, nil)
+		if numDeleted > 0 {
+			o.ui.PrintLinef("Deleted %d older app changes", numDeleted)
+		}
+	}()
+
```

Comment:
app.Delete already deletes app changes (https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/app/recorded_app.go#L362-L365)

Reply:
I believe this is of help when the delete command is executed successfully but is it fair to consider a case when the delete command is getting failed every time it gets called by some script in a certain interval? 
In the case, when delete is not succeeding the above changes can be helpful where it will remove all the app-change once it reaches the threshold.

Reply:
A few thoughts:
- Maybe we should add this change (without the defer) [here](https://github.com/vmware-tanzu/carvel-kapp/blob/1c1ed44974d33259d9f2233d9821a1340edb2ab4/pkg/kapp/cmd/app/delete.go#L157), as this is the only scenario where the deletion of app-changes won't happen.
- Also, it probably doesn't make sense to have this flag (`app-changes-max-to-keep`) for delete command since we are deleting *all* the app changes (in success scenario at least). Instead, we could just maybe keep the last app-change only and delete rest of them.

Reply:
This makes sense to me ^
Does not need to be deferred and the flag seems counter-intuitive 🤔 

However, do we want to have a more conservative default? Maybe keep 5 or even 10 app-changes?
Arguably, we should reduce the default for `kapp deploy` as well but that would require us to look into whether folks are relying on them today.

We often have folks struggling to look at older reconciliations, even though this is much less likely to be useful in the case of delete but it does not hurt to be safer.

Reply:
I have got an interesting thought. If somebody tries to delete based on filters, some resources will be deleted while others will stay. In that case, do we want to remove the history or not?

Reply:
I was thinking we should use `shouldFullyDeleteApp` to check if that's the case.

Reply:
Made the changes to make sure even in case of deleting based on filters we will garbage collect the app-changes once it reaches the threshold (as of now I keep it as 5)